### PR TITLE
More granular reporting for import_medias (fix #275 fix #276)

### LINKED ISSUE #275: The `import_medias` command could be more verbose.
Importing medias should display a summary of the result, something like:

- 3 medias imported flawlessly
- 4 medias import failed (Missing summary 3, path not found 2, etc)

Maybe it should also helps to find the guilty medias by displaying its line number, or the whole line, or something.

### LINKED ISSUE #276: The `import_medias` command should return 1 on errors.
This would help when writing tools around the ideascube commands.

Tested on `ideascube import_medias --dry-run` only, this may apply to other commands, with or without the `--dry-run` option.

### PR DESCRIPTION
**Pull request by yohanboniface**
fix #275
fix #276

### CODE REVIEW

File: ideascube/mediacenter/management/commands/import_medias.py [hunk 1]
```diff
@@ -34,13 +38,26 @@ def abort(self, msg):
...
-                self.stdout.write(u'- {}: {}'.format(key, value))
-        self.stdout.write('-' * 20)
+    def report(self, name, item, level):
+        if level == 1:
+            self.errored = True
+        if name not in self._reports:
+            self._reports[name] = []
+        self._reports[name].append((item, level))
+
+    def reporting(self):
```
Comment (almet) [inline ideascube/mediacenter/management/commands/import_medias.py]: rename this to `output_report` or `display_report`? In general it's easier to understand methods that have a verb in them.

File: ideascube/mediacenter/management/commands/import_medias.py [hunk 2]
```diff
@@ -34,13 +38,26 @@ def abort(self, msg):
...
+            self.stdout.write('# Reports:')
+            for name, items in self._reports.items():
+                self.stdout.write('- {}: {}'.format(name, len(items)))
+                if self.verbosity > 0:
+                    for item, level in items:
+                        if self.verbosity >= level:
+                            out = self.stderr if level == 1 else self.stdout
+                            out.write('  . {}'.format(item))
+        self.stdout.write('\n')
+        if self.errored:
```
Comment (almet) [inline ideascube/mediacenter/management/commands/import_medias.py]: maybe put this in a `self.handle_exit_code` method?

File: ideascube/mediacenter/management/commands/import_medias.py [hunk 3]
```diff
@@ -56,17 +73,21 @@ def handle(self, *args, **options):
...
         rows = self.load(path)
         for row in rows:
             self.add(row)
+        self.reporting()
 
     def add(self, metadata):
         title = metadata.get('title')
         if not title:
-            return self.skip('Missing title', metadata)
+            return self.report('Missing title', metadata, ERROR)
```
Comment (almet) [inline ideascube/mediacenter/management/commands/import_medias.py]: naming: rename to add_error?

**Comment by almet** [comment]
I've added a few comments. In general, why not relying on loggers?

**Comment by almet** [comment]
You could try:

```py
logger = logging.getLogger('main')
logger.debug("message")
```

**Comment by almet** [comment]
Oh, maybe it's a feature to wait the end of all operations to have an output?

**Comment by yohanboniface** [comment]
> Oh, maybe it's a feature to wait the end of all operations to have an output?

Yes, we want to display some "reporting" at the end of the script, with how many success rows, how many errors, etc. And giving more details when running the script with more verbosity.

**Comment by almet** [comment]
That could be great to have a "Reporter" class that you could initialize and then pass to the commands when needed, on which you have different possible methods: "error", "debug", etc. (like the ones from logging), and additionally a "create_report" one, wich outputs the report somewhere (to stdout for now).

What do you think?

**Comment by yohanboniface** [comment]
Yes, sounds good. My plan was actually at some point to have a BaseCommand class that would handle those methods (report, print_report, etc.). Not sure which pattern is better though, but my plan was to wait for a second use case before trying to make this generic. I'll work on that this week-end :)

**Comment by almet** [comment]
I usually tend to [prefer composition over inheritance](https://en.wikipedia.org/wiki/Composition_over_inheritance) :-)

I like to decouple things also because it's simpler to test. If you prefer waiting for a second use case to implement this maybe it's wiser. I'll let you decide !

**Comment by yohanboniface** [comment]
@almet pushed a Report util. May need a bit more use cases to be refined, but that should be a first step :)

**Comment by yohanboniface** [comment]
Example of output:
screenshot from 2016-02-08 16-35-34

**Comment by bochecha** [comment]
Reviewed. :+1:

**Pull request merged**